crates/mpc-protocols/src/mpc_service.rs

```diff
@@ -0,0 +1,115 @@
+use crate::mpc_events::{CreatedProofMPCEvent, MPCEvent};
+use pera_types::base_types::ObjectID;
+use pera_types::event::Event;
+use std::cell::RefCell;
+use std::collections::{HashMap, VecDeque};
+use tokio::sync::{mpsc, Mutex, RwLock};
+
+const MAX_ACTIVE_MPC_INSTANCES: usize = 100;
+
+pub enum MPCInput {
+    InitEvent(CreatedProofMPCEvent),
+    Message,
+    OutputEvent,
+}
+
+struct MPCInstance {
+    status: MPCStatus,
+    /// The channel to send message to this instance's message handler thread when this instance is active
+    messages_handler_sender: Option<mpsc::Sender<MPCInput>>,
```

Comment:
I accepted your suggestion, I'm not a fan of comments either but Zeev asked me to over-document the code so it will be easier for him to grasp it when he comes back. (He's on a holiday rn)